src/biota_models/vegetation/simulation/veg_simulation_2species.py

```diff
@@ -0,0 +1,410 @@
+from abc import ABC, abstractmethod
+from datetime import timedelta
+from pathlib import Path
+from typing import List, Optional, Union
+
+import pandas as pd
+from pydantic import validator
+from tqdm import tqdm
+
+from src.biota_models.vegetation.bio_process.veg_colonisation import Colonization
+from src.biota_models.vegetation.bio_process.veg_hydro_morphodynamics import (
+    Hydro_Morphodynamics,
+)
+from src.biota_models.vegetation.bio_process.veg_mortality import Veg_Mortality
+from src.biota_models.vegetation.model.veg_constants import VegetationConstants
+from src.biota_models.vegetation.model.veg_model import Vegetation
+from src.biota_models.vegetation.output.veg_output_wrapper import VegOutputWrapper
+from src.core import RESHAPE
+from src.core.hydrodynamics.factory import HydrodynamicsFactory
+from src.core.hydrodynamics.hydrodynamic_protocol import HydrodynamicProtocol
+from src.core.simulation.base_simulation import BaseSimulation
+
+
+class _VegetationSimulation_2species(BaseSimulation, ABC):
+    """
+    Implements the `SimulationProtocol`.
```

Comment:
SimulationProtocol only requires one biota. Because you have now a new usecase, you should create a new simulation protocol for that ' MultipleBiotaSimulationProtocol' as well as to create a new base simulation for such case MultipleBiotaBaseSimulation.

I think this requires further work to make it fully functional, but as a first approach is good.

What I would do:
- New Protocols: MultipleBiotaSimulation and BiotaSimulationParameters (which is basically the same as the current SimulationProtocol but excluding the methods, so only properties):
```python

class MultipleBiotaSimulation(Protocol):
  @property
   def biota_simulation_parameters(self) -> List[BiotaSimulationParameters]:
      pass
```
With this you avoid having to put output, output2, veg, veg2, because by accessing the list index you have each of them such as:
```python
# Your current output / veg would be accessed like
self.biota_simulation_parameters[0].output
self.biota_simulation_parameters[0].biota
# Your current output2 / veg2 would be accessed like:
self.biota_simulation_parameters[1].output
self.biota_simulation_parameters[1].biota
```

Reply:
We created the new simulation protocol and the new base simulation, but we did not fully understand how to implement the BiotaSimulationParameters. Therefore, we did not include that yet.